File: 3689-maximum-total-subarray-value-i/3689-maximum-total-subarray-value-i.go
func maxTotalValue(nums []int, k int) int64 {
    if len(nums) <= 1 {
        return int64(0)
    }

    minNum := int(^uint(0) >> 1) // max int
    maxNum := -int(^uint(0) >> 1) - 1 // min int
    for i := 0; i < len(nums); i++ {
        minNum = min(minNum, nums[i])
        maxNum = max(maxNum, nums[i])
    }

    return int64(k * (maxNum - minNum))
}

// array, greedy
// time: O(n)
// space: O(1)

File: 3689-maximum-total-subarray-value-i/3689-maximum-total-subarray-value-i.py
class Solution:
    def maxTotalValue(self, nums: List[int], k: int) -> int:
        min_num, max_num = float("inf"), 0
        for num in nums:
            min_num = min(min_num, num)
            max_num = max(max_num, num)
        
        return k * (max_num - min_num)

# array, greedy
# time: O(n)
# space: O(1)